File: Baekjoon/Greedy/9237.py
# https://www.acmicpc.net/problem/9237

import sys
input = sys.stdin.readline

n = int(input())
tree = sorted(list(map(int, input().split())), reverse = True)

for i in range(n):
    tree[i] += i + 1

print(max(tree) + 1)

File: Baekjoon/Greedy/9237.cpp
// https://www.acmicpc.net/problem/9237

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    vector<int> arr;

    cin >> n;
    arr.resize(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    sort(arr.begin(), arr.end(), greater<int>());
    int result = 0;
    for (int i = 0; i < n; i++) {
        result = max(result, arr[i] + i + 2);
    }

    cout << result << '\n';

    return 0;
}